tuf/api/metadata.py

```diff
@@ -1213,6 +1213,12 @@ def __init__(
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ):
         self.keys = keys
+        if [role for role in set(roles) if role in TOP_LEVEL_ROLE_NAMES]:
+            raise ValueError(
+                "A delegated role can not use top-level role names ("
+                f"{', '.join(TOP_LEVEL_ROLE_NAMES)})"
+            )
```

Comment:
Can you use a shorter exception message?
That's something I have been guilty of on numerous occasions and @jku asked me to refine my messages. :D 
Maybe something like
``` suggestion
            raise ValueError("Delegated roles cannot use top-level role names")
```
An example of a similar message for Root roles where they should be top-level metadata names:
https://github.com/theupdateframework/python-tuf/blob/acb201d6cc685c2560f1000faf47e75db73df7fb/tuf/api/metadata.py#L733

Reply:
Sounds good. What about only showing the role conflicting, such as:
https://github.com/theupdateframework/python-tuf/blob/acb201d6cc685c2560f1000faf47e75db73df7fb/tuf/api/metadata.py#L1231

Reply:
I don't think it's absolutely correct to say that this role is a duplicate.
Yes, we are using the role name for our top-level roles, but it can be understood as if there is another role with the same name inside `Targets.Delegations` which is untrue.
Here the focus is that we don't want delegation role names to be one of the top-level metadata roles as this could lead to unexpected behaviors.